[ERM UI] Update release content page to match mock ups

gherceg opened the pull request:
## Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
Straggler PR I forgot about. My local env is totally borked right now, otherwise I would attach a screenshot. This is mainly wording changes, as well as adding text to the checkbox on the Push Content page. I know I ran this locally when my local env worked fine. I'll add a screenshot once I fix my issue, but wanted to create the PR in the meantime.

<img width="1199" alt="Screen Shot 2021-06-18 at 9 37 03 AM" src="https://user-images.githubusercontent.com/15785053/122569529-d5d17a80-d018-11eb-9794-07046c4472d0.png">

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->
`LINKED_DOMAINS`
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
Will run through QA on parent branch.
### Safety story
<!--
Describe any other pieces to the safety story including
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.
-->
Locally tested at some point. Cannot locally test right now.
### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

gherceg committed:
update text

orangejenny commented on corehq/apps/linked_domain/templates/linked_domain/tabs/push_release_content_tab.html:
```diff
@@ -2,6 +2,14 @@
+  <p>
+    {% blocktrans %}
+      Prepare a new release by selecting which data models you would like to push to the selected downstream project spaces.<br/>
+      <a href="#" target="_blank">Learn more</a> about pushing release content with linked projects.
```
confirming that these docs aren't ready yet?

gherceg replied:
Yeah pretty sure public facing docs are not ready. I'll make a ticket to make sure I address these.

orangejenny commented on corehq/apps/linked_domain/templates/linked_domain/tabs/push_release_content_tab.html:
```diff
@@ -43,14 +51,13 @@
     <div class="form-group">
       <label class="col-sm-3 col-md-2 control-label" for="build-apps">
-        {% trans "Create and release new build for apps" %}
+        {% trans "Re-Build Applications" %}
```
`Re-Build` => `Rebuild`?

orangejenny commented on corehq/apps/linked_domain/templates/linked_domain/tabs/pull_release_content_tab.html:
```diff
@@ -5,7 +5,7 @@
     <p>
       {% blocktrans %}
-        This project is linked to the upstream project <a data-bind="attr: {'href': domainLink.upstreamUrl}, text: domainLink.master_domain"></a>.<br/>
+        This project is linked to the upstream project <a data-bind="attr: {'href': domainLink.upstreamUrl}, text: domainLink.upstreamDomain"></a>.<br/>
```
Is this change supposed to be part of this PR? I looked at the version of `domain_links.js` that's on the `gh/improve-erm-ui` which is [this](https://github.com/dimagi/commcare-hq/blob/e83c1984be750d0499a7dd4ddd8be9ea94a3013b/corehq/apps/linked_domain/static/linked_domain/js/domain_links.js), and it still has `master_domain`.

gherceg replied:
Good catch. I messed up a conflict resolution and thought I had taken care of it, but obviously not.

orangejenny reviewed (approved):
Fine to merge if the `upstreamDomain` change is intended.

gherceg committed:
address feedback

gherceg committed:
replace release with push

gherceg commented:
Updated with screenshot and added another commit to replace "release" with "push". If we stick with this wording I will rename the underlying variables to match the updated lingo.
